topology/controllers.go

```diff
@@ -0,0 +1,111 @@
+package topology
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"text/template"
+
+	"github.com/flanksource/commons/logger"
+	"github.com/flanksource/incident-commander/api"
+	babel "github.com/jvatic/goja-babel"
+	"github.com/labstack/echo/v4"
+)
+
+var jsComponentTpl *template.Template
+
+func init() {
+	tpl, err := template.New("registry").Parse(jsComponentRegistryTpl)
+	if err != nil {
+		logger.Fatalf("error parsing template 'jsComponentRegistryTpl'. %v", err)
+	}
+
+	jsComponentTpl = tpl
+}
+
+type component struct {
+	Name string
+	JS   string
+}
+
+// GetCustomRenderer returns an application/javascript HTTP response
+// with custom components and a registry.
+// This registry needs to be used to select custom components
+// for rendering of properties and cards.
+func GetCustomRenderer(ctx echo.Context) error {
+	// 1. Read the template of the topology
+	sysTplID := ctx.QueryParams().Get("id")
```

Comment:
from a code style perspective, comments should say why, not what. Variable names should not be abbreviated except for the few common e.g. id
```suggestion
	id := ctx.QueryParams().Get("id")
```

Reply:
I agree.
 
 It's just a convention I'm used to following where I write the steps before I start writing the code.
 Will remove it.